src/enrichment_tables/geoip.rs

```diff
@@ -0,0 +1,413 @@
+use std::{collections::BTreeMap, fs, net::IpAddr, sync::Arc, time::SystemTime};
+
+use enrichment::{Case, Condition, IndexHandle, Table};
+use serde::{Deserialize, Serialize};
+use value::Value;
+
+use crate::config::{EnrichmentTableConfig, EnrichmentTableDescription, GenerateConfig};
+
+// MaxMind GeoIP database files have a type field we can use to recognize specific
+// products. If we encounter one of these two types, we look for ASN/ISP information;
+// otherwise we expect to be working with a City database.
+const ASN_DATABASE_TYPE: &str = "GeoLite2-ASN";
+const ISP_DATABASE_TYPE: &str = "GeoIP2-ISP";
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub struct GeoipConfig {
+    pub path: String,
+    #[serde(default = "default_locale")]
+    pub locale: String,
+}
+
+// valid locales are: “de”, "en", “es”, “fr”, “ja”, “pt-BR”, “ru”, and “zh-CN”
+//
+// https://dev.maxmind.com/geoip/docs/databases/city-and-country?lang=en
+//
+// TODO try to determine the system locale and use that as default if it matches a valid locale?
+fn default_locale() -> String {
+    "en".to_string()
+}
+
+impl GenerateConfig for GeoipConfig {
+    fn generate_config() -> toml::Value {
+        toml::Value::try_from(Self {
+            path: "/path/to/GeoLite2-City.mmdb".to_string(),
+            locale: default_locale(),
+        })
+        .unwrap()
+    }
+}
+
+inventory::submit! {
+    EnrichmentTableDescription::new::<GeoipConfig>("geoip")
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "geoip")]
+impl EnrichmentTableConfig for GeoipConfig {
+    async fn build(
+        &self,
+        _: &crate::config::GlobalOptions,
+    ) -> crate::Result<Box<dyn Table + Send + Sync>> {
+        Ok(Box::new(Geoip::new(self.clone())?))
+    }
+}
+
+#[derive(Clone)]
+pub struct Geoip {
+    config: GeoipConfig,
+    dbreader: Arc<maxminddb::Reader<Vec<u8>>>,
+    last_modified: SystemTime,
+    indexes: Vec<(Case, Vec<String>)>,
```

Comment:
```suggestion

```

The indexes are not being used in the lookup, so we should be safe to just not store them at all.

Reply:
That may be the case at the moment, but if we are going to future proof `find_table_row` then it makes sense to also future proof this.